packages/mask/src/plugins/Savings/protocols/YearnProtocol.ts

```diff
@@ -0,0 +1,146 @@
+import type Web3 from 'web3'
+import type { AbiItem } from 'web3-utils'
+import BigNumber from 'bignumber.js'
+import { ZERO } from '@masknet/web3-shared-base'
+import { ChainId, createContract, FungibleTokenDetailed } from '@masknet/web3-shared-evm'
+
+import { ProtocolType, SavingsProtocol } from '../types'
+import type { ERC20 } from '@masknet/web3-contracts/types/ERC20'
+import ERC20ABI from '@masknet/web3-contracts/abis/ERC20.json'
+import { VaultInterface, Vault, Yearn } from '@yfi/sdk'
+
+export class YearnProtocol implements SavingsProtocol {
+    static DEFAULT_APR = '0.1'
+
+    private _apr = '0.00'
+    private _balance = ZERO
+
+    constructor(readonly pair: [FungibleTokenDetailed, FungibleTokenDetailed]) {}
+
+    get type() {
+        return ProtocolType.YEARN
+    }
+
+    get apr() {
+        return this._apr
+    }
+
+    get balance() {
+        return this._balance
+    }
+
+    get bareToken() {
+        return this.pair[0]
+    }
+
+    get stakeToken() {
+        return this.pair[1]
+    }
+
+    public async updateApr(chainId: ChainId, web3: Web3) {
+        try {
+            // @ts-ignore
+            const yearn = new Yearn(chainId, {
```

Comment:
suggest, provide new typing for yearn

```
declare module "@yfi/sdk" {
  export class Yearn { /** ... new typing ... */ }
}
```

see https://www.typescriptlang.org/docs/handbook/declaration-merging.html

Reply:
not sure i understand